File: java/src/main/java/it/giannibombelli/wsc2026/payment/domain/payment/TransactionStatus.java
package it.giannibombelli.wsc2026.payment.domain.payment;

public enum TransactionStatus {
    STARTED("started"),
    ACCEPTED("accepted"),
    REJECTED("rejected"),
    REFUNDED("refunded");

    private final String value;

    TransactionStatus(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: typescript/src/payment/domain/payment/transactionStatus.ts
import { requireArgument } from '@/common/utils/requireArgument.js';
// Stati di una Transaction.

export const TransactionStatus = {
  STARTED: 'STARTED',
  ACCEPTED: 'ACCEPTED',
  REJECTED: 'REJECTED',
  REFUNDED: 'REFUNDED',
} as const;

export type TransactionStatus = (typeof TransactionStatus)[keyof typeof TransactionStatus];

export function transactionStatusFromLabel(label: string): TransactionStatus {
  requireArgument(label, 'transaction status label');
  const value = Object.values(TransactionStatus).find((status) => status === label);
  if (value === undefined) {
    throw new Error(`unknown transaction status: ${label}`);
  }
  return value;
}
